tests/unit/services/intl-test.js

```diff
@@ -238,4 +240,16 @@ module('service:intl', function(hooks) {
     assert.expect(1);
     assert.equal(this.intl.exists('bar'), false);
   });
+
+  test('changing the locale emits the `localeChanged` event and the new locale is available', async function(assert) {
+    assert.expect(1);
+    
+    const newLocale = ['de', 'en-us'];
+
+    addListener(this.intl, 'localeChanged', () => {
+      assert.deepEqual(get(this.intl, 'locale'), newLocale);
```

Comment:
@buschtoens this gets called twice where I'd expect it to be called once.  Could be an artifact of setting the locale during the test setup, didn't get a chance to pull down the branch and investigate.